session_3/blog/app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,89 @@
+class ArticlesController < ApplicationController
+  load_and_authorize_resource
+
+  before_action :set_article, only: [:show, :edit, :update, :destroy]
+
+  # GET /articles
+  # GET /articles.json
+  def index
+    #unregistered users can only read 5 articles
+    if session[:user_id] && !session[:private_articles_left]
+      session[:private_articles_left]=5
+    end
+    @articles = Article.all
+  end
+
+  # GET /articles/1
+  # GET /articles/1.json
+  def show
+    #to keep count of private articles read
+    if session[:private_articles_left] == 0
+      render "articles/limit"
+    elsif session[:user_id] && !Article.find(params[:id]).public && !session[:admin]
```

Comment:
- You don't need to do `Article.find(params[:id])`, it's loaded by default in `@article` because of the `set_article` before action.
- You don't need `session[:admin]` and can use `current_user.admin` if you have added the column.